### PR TITLE
Feat feed page

### PR DESCRIPTION
**Pull request by ashitha-18**
created an endpoint for feed page that can 
- retrieve paginated data
- accept search query (nameOfWork)
- accept filter (medium, licenseType)

### CODE REVIEW

**Comment by ashitha-18** [comment]
Added features
- searching based on nameOfCreator
- filter based on date

File: src/services/feedPage.services.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { User } from "../models/user.model"
+import { executeUserOp, getPartialUserOp } from "../integrations/biconomy";
+import { ethers } from "ethers";
+import { getNetworkConfig } from "../utils/network.utils";
+import { HttpException } from "../utils/exceptions.utils";
+import { BAD_REQUEST, ONE, ZERO } from "../utils/constants.utils";
+import { ICreateWorkSchema, ICreatorWorkMetadata } from "../models/interfaces.models";
+
+class FeedPageService {
+    getPaginatedWorks = async (page: number, count: number, nameOfWork: string, nameOfCreator: string, medium: string, licenseType: string, startDate: string, endDate: string) => {
```
Comment (ogubuikeAlex) [inline src/services/feedPage.services.ts]: Remember there is no way the user will specify what they are sending to the back end.

Its a single search bar, so the user simply comes there to type a search query and that query could either be a creators name or name of a created work.

On our end, we should be checking

1. If we have we have a creator whose name includes the search query
2.  If we have we have any created work whose name includes the search query

Based on this we do not need to have `nameOfCreator` and `nameOfWork`, we should just have a `searchQuery` that will represent whatever the user has typed in.

-----

This is also lot of input parameters and it we could easily make a mistake while collecting. So declare a type and pass the type here instead

For example:
```typescript
type GetPaginatedWorksRequest = {
  page?: number,
  count?: number
  searchParam?: string,
  medium?: string,
  licenseType?: string,
  startDate?: string,
  endStart?: string
}
```

getPaginatedWorks = async (request: GetPaginatedWorksRequest){}

Comment (ogubuikeAlex) [reply src/services/feedPage.services.ts]: Remember if I do not pass in any of these parameters, the endpoint should not fail

Instead it should return all items with a default pagination (eg 10)

So if I call this endpoint with no extra information, as long as there is data in the DB I should get the first ten items from the DB

File: src/services/feedPage.services.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if (startDate || endDate) {
+            query.timeStamp = {};
+            if (startDate) {
+                query.timeStamp.$gte = new Date(startDate);
+            }
+            if (endDate) {
+                query.timeStamp.$lte = new Date(endDate);
+            }
+        }
+        const works = await CreatedWork.find(query).skip(skip).limit(count);
```
Comment (ogubuikeAlex) [inline src/services/feedPage.services.ts:89]: What if query is an empty object?

**Review by ogubuikeAlex — CHANGES_REQUESTED** [review]

File: src/services/feedPage.services.ts (hunk 1 above)
Comment (ashitha-18) [reply src/services/feedPage.services.ts]: This is already satisfied as query is initialized to { } and find( { } ) returns all the items from DB with a default pagination

Comment (ashitha-18) [reply src/services/feedPage.services.ts]: Made changes to how search query is checked.
Declared type GetPaginatedWorksRequest.

File: src/services/feedPage.services.ts (hunk 2 above)
Comment (ashitha-18) [reply src/services/feedPage.services.ts:89]: it returns all the items from DB with a default pagination

**Pull request merged**